switchtec.c

```diff
@@ -150,8 +150,10 @@ static void mrpc_cmd_submit(struct switchtec_dev *stdev)
 	schedule_delayed_work(&stdev->mrpc_timeout,
 			      msecs_to_jiffies(500));
 
-	if (stdev->dma_mrpc)
+	if (stdev->dma_mrpc) {
 		stdev->dma_mrpc->status = SWITCHTEC_MRPC_STATUS_INPROGRESS;
+		memset(stdev->dma_mrpc->data, 0xFF, SWITCHTEC_MRPC_PAYLOAD_SIZE);
```

Comment:
You almost certainly want to do this before you submit the command to the firmware... If the CPU gets rescheduled,  you might end up clearing the buffer after the DMA writes to it.

Reply:
Both of the status and dma buffer init should be before fire the mrpc cmd.